File: picture-server/src/test/java/com/applory/pictureserver/RestTemplateInterceptor.java
package com.applory.pictureserver;

import org.springframework.http.HttpHeaders;
import org.springframework.http.HttpRequest;
import org.springframework.http.client.ClientHttpRequestExecution;
import org.springframework.http.client.ClientHttpRequestInterceptor;
import org.springframework.http.client.ClientHttpResponse;

import java.io.IOException;

public class RestTemplateInterceptor implements ClientHttpRequestInterceptor {

    private final String token;

    public RestTemplateInterceptor(String token) {
        this.token = token;
    }

    @Override
    public ClientHttpResponse intercept(HttpRequest request, byte[] body, ClientHttpRequestExecution execution) throws IOException {
        HttpHeaders headers = request.getHeaders();
        if (!headers.containsKey("Authorization")) {
            headers.set("Authorization", "Bearer " + token);
        }

        return execution.execute(request, body);
    }
}

File: picture-server-kt/src/test/kotlin/com/applory/pictureserverkt/RestTemplateInterceptor.kt
package com.applory.pictureserverkt

import org.springframework.http.HttpRequest
import org.springframework.http.client.ClientHttpRequestExecution
import org.springframework.http.client.ClientHttpRequestInterceptor
import org.springframework.http.client.ClientHttpResponse

class RestTemplateInterceptor(val token: String): ClientHttpRequestInterceptor {
    override fun intercept(
        request: HttpRequest,
        body: ByteArray,
        execution: ClientHttpRequestExecution
    ): ClientHttpResponse {
        val headers = request.headers
        if (!headers.containsKey("Authorization")) {
            headers.set("Authorization", "Bearer $token")
        }

        return execution.execute(request, body)
    }

}
